// Copyright 2015 gRPC authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

message ClientRequest {
    string interface = 1;
    int32 money = 2;
    int32 id = 3;
    repeated int32 writeidset = 4;
}

message BranchResponse {
    string interface = 1;
    int32 money = 2;
    repeated int32 writeidset = 3;
}

service Branch {
    // Send a client request message to the bank branch 
    rpc MsgDelivery(ClientRequest) returns (BranchResponse) {}
    
    // Send an update request message to the peer branches
    rpc MsgPropagation(ClientRequest) returns (BranchResponse) {}

}

